src/google-social-provider.js

```diff
@@ -7,105 +7,59 @@ GoogleSocialProvider = function(dispatchEvent) {
- */
-GoogleSocialProvider.prototype.loadContacts_ = function() {
-  // TODO: should we periodically check for new friends?  Or just force
-  // users to logout then login again to detect new friends?
-  this.googleGet_('plus/v1/people/me/people/visible').then(function(resp) {
-    for (var i = 0; i < resp.items.length; ++i) {
-      var friend = resp.items[i];
-      this.addUserProfile_({
-        userId: friend.id,
-        name: friend.displayName,
-        imageData: friend.image.url,
-        url: friend.url
-      });
-    }
-  }.bind(this)).catch(function(e) {
-    this.logger.error('Error loading Google users', e);
-  }.bind(this));
-};
-
-/*
  * Returns UserProfile object for the logged in user.
  */
-GoogleSocialProvider.prototype.getMyUserProfile_ = function() {
+GoogleSocialProvider.prototype.getMyImage_ = function() {
   if (!this.loginState_) {
-    throw 'Error in GoogleSocialProvider.getMyUserProfile_: not logged in';
+    throw 'Error in FacebookSocialProvider.getMyUserProfile_: not logged in';
   }
-  var cachedUserProfile =
-      this.loginState_.authData[this.networkName_].cachedUserProfile;
-  return {
-    userId: this.getUserId_(),
-    name: cachedUserProfile.name,
-    lastUpdated: Date.now(),
-    url: cachedUserProfile.link,
-    imageData: cachedUserProfile.picture + '?sz=50'
-  };
+  return this.loginState_.authData[this.networkName_].cachedUserProfile
+      .picture + '?sz=50';
 };
 
 /*
  * Makes get request to Google endpoint, and returns a Promise which
  * fulfills with the response object.
  */
-GoogleSocialProvider.prototype.googleGet_ = function(endPoint) {
+GoogleSocialProvider.prototype.googlePost_ = function(endPoint, data) {
   if (!this.loginState_) {
     throw 'Not signed in';
   }
   var xhr = new XMLHttpRequest();
   var url = 'https://www.googleapis.com/' + endPoint +
-      '?key=AIzaSyA1Q7SiEeUdSJwansl2AUFXLpVdnsXUzYg';
-  xhr.open('GET', url);
+      '?key=AIzaSyA1Q7SiEeUdSJwansl2AUFXLpVdnsXUzYg&alt=json';
+  xhr.open('POST', url);
   xhr.setRequestHeader(
       'Authorization',
       'Bearer ' + this.loginState_.authData.google.accessToken);
+  xhr.setRequestHeader(
+      'Content-Type', 'application/json');
   return new Promise(function(fulfill, reject) {
-    // TODO: error checking
     xhr.onload = function() {
       fulfill(JSON.parse(this.response));
     };
-    xhr.send();
+    xhr.onerror = function(e) {
+      reject('Error posting to Google ' + e);
+    };
+    xhr.send(data);
   });
 };
 
+GoogleSocialProvider.prototype.sendEmail = function(to, subject, body) {
+  if (!this.loginState_) {
+    throw 'Not signed in';
+  }
+  var email ='"Content-Type: text/plain; charset="us-ascii"\n' +
+      'MIME-Version: 1.0\n' +
+      'Content-Transfer-Encoding: 7bit\n' +
+      'to: ' + to + '\n' +
+      'from: ' + this.loginState_.authData[this.networkName_].email + '\n' +
+      'subject: ' + subject + '\n\n' + body;
+  this.googlePost_('gmail/v1/users/me/messages/send',
```

Comment:
I think you can just return this.googlePost_ (which could also propagate a failed post)